src/lib/autoinstall/ask/profile_reader.rb

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "autoinstall/ask/dialog"
+require "autoinstall/ask/question"
+require "autoinstall/ask/question_option"
+require "autoinstall/script"
+
+module Y2Autoinstall
+  module Ask
+    # This class turns an <ask-list> section into a list of Ask::Dialog objects
+    class ProfileReader
+      # Constructor
+      #
+      # @param ask_list_section [AskListSection] <ask-list> section from the profiel
+      # @param stage [Symbol] :initial or :cont
```

Comment:
I would mention here that it stage means if it is first stage or second ( cont is second )